runtime/src/accounts_background_service.rs

```diff
@@ -169,23 +174,60 @@ impl SnapshotRequestHandler {
                     shrink_time.stop();
                 }
 
+                let block_height = snapshot_root_bank.block_height();
+                let snapshot_type = if block_height
+                    % self.snapshot_config.full_snapshot_archive_interval_slots
+                    == 0
+                {
+                    Some(SnapshotType::FullSnapshot)
+                } else if block_height
+                    % self
+                        .snapshot_config
+                        .incremental_snapshot_archive_interval_slots
+                    == 0
+                {
+                    Some(SnapshotType::IncrementalSnapshot(last_full_snapshot_slot))
+                } else {
+                    None
+                };
```

Comment:
Now AccountsBackgroundService is the one responsible for setting the snapshot type (previously this was in AccountsHashVerifier).

Note that the `None` here is for AccountsPackages that will _not_ be archived, which is possible if the incremental snapshot interval does _not_ equal the accounts hash interval.

Reply:
Yeah using the `block_height` is a lot simpler than trying to find the first slot that crossed the full snapshot boundary. Interesting bit here is the full snapshots will be correlated with skip rate, i.e. the higher the skip rate, the less often - in wallclock time - we will take full snapshots. I think this is fine for v1.

Reply:
Ah! That is interesting. Good to note.